### PR TITLE
Bootstrap runner

### PR DESCRIPTION
**Pull request by edjroz**
Solves #13

### CODE REVIEW

**Review by oten91 — APPROVED** [review]

File: run.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+			if currentUpgrade == nil {
+				// wait for upgrade if no current upgrade this way the blockHeight won't change
+				currentUpgrade = <-upgrades
+			}
+			upgrade := currentUpgrade
+			headerEvt := rawHeaderEvt.Data.(tmTypes.EventDataNewBlockHeader)
+			if upgrade.Height != headerEvt.Header.Height {
+				continue
+			}
+			if err := cmd.Process.Kill(); err != nil { // PROCESS MUST DIE BEFORE UPGRADING; cfg.Current is a symlink otherwise bugs might happen
```
Comment (andrewnguyen22) [inline run.go:92]: To be clear, the process kills itself at that block height @ BeginBlock()

File: run.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		case <-ctx.Done():
+			if err := cmd.Process.Kill(); err != nil {
+				errors <- err
+			}
+			return // singal to kill process was sent in case it was sent inside a particular upgrade terminate execution
+		}
+	}
+}
+
+// WaitForUpgrade listens transactions and filters upgrades, passess them to the upgrade cahnnel
```
Comment (andrewnguyen22) [inline run.go]: upgrade `channel`

File: run.go (hunk 1 above)
Comment (edjroz) [reply run.go:92]: It's ok, this is a safeguard due to the usage of symlinks, if the process is already dead go does not throw an error or panic.

File: run.go (hunk 2 above)
Comment (edjroz) [reply run.go]: fixed.

File: run.go (hunk 1 above)
Comment (andrewnguyen22) [reply run.go:92]: Perf

**Review by andrewnguyen22 — APPROVED** [review]

**Pull request merged**